homeassistant/components/waterkotte_heatpump/__init__.py

```diff
@@ -0,0 +1,88 @@
+"""The Waterkotte Heatpump integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from typing import Any
+
+from pywaterkotte.ecotouch import Ecotouch, TagData
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.typing import StateType
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+UPDATE_INTERVAL = 10
+
+
+class EcotouchCoordinator(DataUpdateCoordinator[dict[TagData, Any]]):
+    """heatpump coordinator."""
+
+    def __init__(self, heatpump: Ecotouch, hass: HomeAssistant) -> None:
+        """Init coordinator."""
+        self._heatpump = heatpump
+
+        self.alltags: set[TagData] = set()
+
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=timedelta(seconds=UPDATE_INTERVAL),
+        )
+
+    async def _async_update_data(self) -> dict[TagData, Any]:
+        """Fetch the latest data from the source."""
+        tag_list = list(self.alltags)
+        return await self.hass.async_add_executor_job(
+            self._heatpump.read_values, tag_list
+        )
+
+    def get_tag_value(self, tag: TagData) -> StateType:
+        """Return a tag value."""
+        return self.data.get(tag, None)
+
+    @property
+    def heatpump(self) -> Ecotouch:
+        """Heatpump api."""
+        return self._heatpump
+
```

Comment:
Put Coordinator in its own Coordinator.py file

Reply:
redolved with [eab7ef6](https://github.com/home-assistant/core/pull/88140/commits/eab7ef65b5f3a23e0837922b0fbdb496ad21f652)